pkg/config/mesh/networks_watcher.go

```diff
@@ -115,19 +125,50 @@ func (w *internalNetworkWatcher) SetNetworks(meshNetworks *meshconfig.MeshNetwor
 		// Store the new config.
 		w.prevNetworks = w.networks
 		w.networks = meshNetworks
-		handlers = append([]func(){}, w.handlers...)
+		handlers = append([]*WatcherHandler{}, w.handlers...)
 	}
 	w.mutex.Unlock()
 
 	// Notify the handlers of the change.
 	for _, h := range handlers {
-		h()
+		h.Handler()
 	}
 }
 
 // AddNetworksHandler registers a callback handler for changes to the mesh network config.
-func (w *internalNetworkWatcher) AddNetworksHandler(h func()) {
+func (w *internalNetworkWatcher) AddNetworksHandler(h *WatcherHandler) {
 	w.mutex.Lock()
 	defer w.mutex.Unlock()
 	w.handlers = append(w.handlers, h)
 }
+
+// DeleteNetworksHandler deregister a callback handler for changes to the mesh network config.
+func (w *internalNetworkWatcher) DeleteNetworksHandler(h *WatcherHandler) {
+	if h == nil {
+		return
+	}
+
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+
+	if len(w.handlers) == 0 {
+		return
+	}
+
+	index, ok := getWatcherHandlerIndex(h, w.handlers)
```

Comment:
nit: can simplify with 
```go
slices.FilterInPlace(w.handlers, func(i) bool { return i == h }
```

Reply:
Updated. But I'm little confused whether the underlaying slice will still hold old object or not.

When we want to remove object from the slice, we always assign nil to object explicitly.
```
slice[index] = nil 
slice = append(slice[:index], slice[index+1:]...)
```

Reply:
Oh. Link pr https://github.com/istio/istio/pull/48434.